### DISCUSSION TITLE
build.env for Preview via cli or NPM_RC setup with npm and github

### DISCUSSION THREAD

**Discussion by ralphilius** [vercel/vercel#5013] [Help] [answered]
Hello,

How should I add a variable environment for my Preview environment via cli? I want cli because I need to set a NPM_RC variable from a file, and the UI won't have line break. I try with setting build.env in vercel.json/now.json, but received this error:

> Couldn't find package @scope/xxx on the "npm" registry

**Comment by styfle** [comment, marked as answer]
Hi @ralphilius 

It sounds like the variable wasn't assigned or the token was incorrect.

See this guide for steps to setup private npm dependencies with Vercel:

https://vercel.com/knowledge/using-private-dependencies-with-vercel
